packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4283,21 +4289,46 @@ export class ContainerRuntime
 				} = {};
 				const trace = Trace.start();
 
-				const versions = await this.storage.getVersions(
-					null,
-					1,
-					"prefetchLatestSummaryBeforeClose",
-					FetchSource.noCache,
-				);
-				assert(!!versions && !!versions[0], 0x137 /* "Failed to get version from storage" */);
-				stats.getVersionDuration = trace.trace().duration;
+				let maybeSnapshotTree: ISnapshotTree | null;
+				const scenarioName = "prefetchLatestSummaryBeforeClose";
+				// If loader supplied us the ISnapshot when loading, the new getSnapshotApi is supported and feature gate is ON, then use the
+				// new API, otherwise it will reduce the service performance because the service will need to recalculate the full snapshot
+				// in case previously getSnapshotApi was used and now we use the getVersions API.
+				if (
+					this.isSnapshotInstanceOfISnapshot &&
+					this.storage.getSnapshot !== undefined &&
+					this.mc.config.getBoolean("Fluid.Container.UseLoadingGroupIdForSnapshotFetch2") ===
+						true
+				) {
+					const snapshot = await this.storage.getSnapshot({
```

Comment:
Curious, is this change required?
It looks like nice to have, in a sense that it helps with future code cleanup (once we start removing old APIs).
There is no sensitivity for perf from client side here (this is not on critical path).
COGS savings for SPO are likely a plus.

Just trying to understand the motivation...

Reply:
@vladsud I made this change thinking that it will affect the load performance for normal clients, no? Because spo would have cached the partial snapshot but when summarizer will hit the old path to fetch full snapshot, the spo will then cache this full snapshot, and then other clients will request a partial snapshot and spo will recalculate it?